packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -51,11 +51,48 @@
   }
 }
 
-@mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
+@mixin iui-dialog-position {
+  top: 33%;
+  left: 50%;
+  transform: translate(-50%, -33%);
+
+  &[data-position] {
+    position: fixed;
+    top: auto;
+    left: auto;
+    transform: none;
```

Comment:
This looks a bit random. Any reason you added `position: fixed` here instead of to the base dialog? And `top: auto`/`left: auto` could also be moved into the individual placement cases where you know the exact value.

Reply:
They're needed in every iteration of the placement, including the centering. Honestly I placed them in there because that was the only solution I was able to get working for a default center.